server/src/Domain/Entities/FlashcardSet.cs

```diff
@@ -1,11 +1,19 @@
 using StudyZen.Domain.Enums;
+using StudyZen.Domain.Constraints;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.ComponentModel.DataAnnotations;
 
 namespace StudyZen.Domain.Entities;
 
 public sealed class FlashcardSet : BaseEntity
 {
     public int? LectureId { get; set; }
```

Comment:
https://www.entityframeworktutorial.net/code-first/foreignkey-dataannotations-attribute-in-code-first.aspx